lib/promise.rb

```diff
@@ -19,7 +19,7 @@ def self.resolve(obj)
   end
 
   def self.all(enumerable)
-    Group.new(enumerable).promise
+    Group.new(new, enumerable).promise
```

Comment:
I'd be tempted to use `self.new` here just to make the syntax a little less weird, but it's also entirely unnecessary, so your call